Client/Frontend/Browser/BrowserViewController.swift

```diff
@@ -1451,7 +1471,22 @@ extension BrowserViewController: URLBarDelegate {
             // Leave some space near the bottom for easier dismissal
             popover.outerMargins.bottom = 80.0
         }
-        popover.present(from: urlBar.locationView.shieldsButton, on: self)
+        popover.present(from: urlBar.shieldsButton, on: self)
+    }
+    
+    func urlBarDidTapMenuButton(_ urlBar: URLBarView) {
+        guard let selectedTab = tabManager.selectedTab else { return }
+        
+        let homePanel = HomeMenuController(profile: profile, tabState: selectedTab.tabState)
+        homePanel.preferredContentSize = CGSize(width: PopoverController.preferredPopoverWidth, height: 600.0)
```

Comment:
I don't agree with extracting all hardcoded values but why is there a `preferredPopoverWidth` but height is hardcoded?

Reply:
Hmm my thought process (I think) when doing this was: Design has made these popups mostly all the same size width, but sometimes they have different heights based on their content.

This isn't the case with the current 2 popovers due to porting the old screens. I _could_ make `preferredPopoverHeight` or change it to `preferredPopoverSize` but I'm not sure how much it'll be used

Reply:
Got it, fine by me